mirascope/core/vertex/call_response_chunk.py

```diff
@@ -0,0 +1,82 @@
+"""This module contains the `VertexCallResponseChunk` class.
+
+usage docs: learn/streams.md#handling-streamed-responses
+"""
+
+from vertexai.generative_models import FinishReason, GenerationResponse
+
+from ..base import BaseCallResponseChunk
+
+
+class VertexCallResponseChunk(
+    BaseCallResponseChunk[GenerationResponse, FinishReason]  # , Candidate.FinishReason]
+):
+    """Convenience wrapper around chat completion streaming chunks.
+
+    When using Mirascope's convenience wrappers to interact with Vertex models via
+    `VertexCall`, responses using `VertexCall.stream()` will return a
+    `VertexCallResponseChunk`, whereby the implemented properties allow for simpler
+    syntax and a convenient developer experience.
+
+    Example:
+
+    ```python
+    from mirascope.core import prompt_template
+    from mirascope.core.vertex import vertex_call
+
+
+    @vertex_call("vertex-1.5-flash", stream=True)
+    @prompt_template("Recommend a {genre} book")
+    def recommend_book(genre: str):
+        ...
+
+
+    stream = recommend_book("fantasy")  # response is an `VertexStream`
+    for chunk, _ in stream:
+        print(chunk.content, end="", flush=True)
+    ```
+    """
+
+    @property
+    def content(self) -> str:
+        """Returns the chunk content for the 0th choice."""
+        return self.chunk.candidates[0].content.parts[0].text
+
+    @property
+    def finish_reasons(self) -> list[FinishReason]:
+        """Returns the finish reasons of the response."""
+        return [candidate.finish_reason for candidate in self.chunk.candidates]
+
+    @property
+    def model(self) -> None:
+        """Returns the model name.
+
+        google.generativeai does not return model, so we return None
+        """
+        return None
+
+    @property
+    def id(self) -> str | None:
+        """Returns the id of the response.
+
+        google.generativeai does not return an id
+        """
+        return None
+
+    @property
+    def usage(self) -> None:
+        """Returns the usage of the chat completion.
+
+        google.generativeai does not have Usage, so we return None
+        """
+        return None
```

Comment:
we should check if you can track usage when streaming with gemini. if so, we should implement that here

Reply:
I can't get the tokens count for streaming.

<img width="632" alt="image" src="https://github.com/user-attachments/assets/47cbd995-433a-4970-bcd3-9e84373bb330">

Reply:
ah ok no worries then